services/httpd/handler.go

```diff
@@ -1087,13 +1089,23 @@ func gzipFilter(inner http.Handler) http.Handler {
 			inner.ServeHTTP(w, r)
 			return
 		}
-		gz := gzip.NewWriter(w)
-		defer gz.Close()
+		gz := gzipWriterPool.Get().(*gzip.Writer)
+		gz.Reset(w)
+		defer func() {
+			gz.Close()
+			gzipWriterPool.Put(gz)
```

Comment:
Possibly consider resetting to `ioutil.Discard` here to remove the lingering reference to the `ResponseWriter`? This would also ensure that all writers returned from `Get()` are in the same state.